src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -20,15 +20,48 @@
 
 namespace Stratis.Bitcoin.Features.Miner
 {
+    /// <summary>
+    /// <see cref="PosMinting"/> is used in order to generate new blocks. It involves a sort of lottery, similar to proof-of-work,
+    /// but the chances of winning this lottery is proportional to how many coins you are staking, not on hashing power.
+    /// </summary>
+    /// <remarks>
+    /// Staking is attempted only if the node is fully synchronized and connected to the network.
+    /// If not it will wait  till node is synced. Only transactions that were confirmed at least
+    /// <see cref="PosConsensusOptions.StakeMinConfirmations"/> blocks ago are eligible for staking.
+    /// </remarks>
+    /// <remarks>
+    /// The overall process for "attempting" to mine a PoS block looks like this:
+    /// 1) Create new block with transactions from <see cref="TxMempool"/>
+    /// 2) Get UTXOs (Unspent Transaction Output) that can participate in staking (have suitable depth).
+    /// 3) Split those UTXOs in subsets so each subset contains max of <see cref="utxoStakeDescriptionsPerCoinstakeWorker"/> transactions
+    /// and run CoinstakeWorker for each subset. Splitting is done only to achieve a good level of parallelism.
+    /// 4) CoinstakeWorker will try different versions of a block we are trying to mine (the only difference is the
+    /// timestamp which is tried in a time interval from now to now - searchInterval seconds.
+    /// Search interval is a length of an unexplored block time space in seconds.
+    /// Worker calculates the kernel's hash using the next formula:
+    /// hash(stakeModifierV2 + stakingCoins.Time + prevout.Hash + prevout.N + transactionTime).
+    /// Then it calculates staking target using the next formula:
+    /// (header of a block that we are trying to mine as BigInteger * amount of satoshi we are staking).
+    /// We compare kernel's hash (as BigInteger) against the staking target, if it's greater then we met the criteria
+    /// and kernel is found.
+    /// So the more coins we stake the higher the staking target => higher chances of meeting the criteria.
```

Comment:
avoid `=>`, use english, in non-formulas ... `and so the higher the chance is to meet the criteria.`